packages/teams-js/test/private/externalAppCardActions.spec.ts

```diff
@@ -150,6 +183,39 @@ describe('externalAppCardActions', () => {
           }
           return expect(promise).rejects.toEqual(testError);
         });
+        it(`should throw error on invalid app ID if it contains script tag with context - ${frameContext}`, async () => {
+          expect.assertions(1);
+          await utils.initializeWithContext(frameContext);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { externalAppCardActions: {} } });
+          const invalidAppId = 'invalidAppIdwith<script>alert(1)</script>';
+          try {
+            externalAppCardActions.processActionOpenUrl(invalidAppId, testUrl);
+          } catch (e) {
+            expect(e).toEqual(new Error('App ID is not valid.'));
+          }
+        });
+        it(`should throw error on invalid app ID if it contains non printabe ASCII characters with context - ${frameContext}`, async () => {
+          expect.assertions(1);
+          await utils.initializeWithContext(frameContext);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { externalAppCardActions: {} } });
+          const invalidAppId = 'appId\u0000';
+          try {
+            externalAppCardActions.processActionOpenUrl(invalidAppId, testUrl);
+          } catch (e) {
+            expect(e).toEqual(new Error('App ID is not valid.'));
+          }
+        });
+        it(`should throw error on invalid app ID if its size exceeds 256 characters with context - ${frameContext}`, async () => {
+          expect.assertions(1);
+          await utils.initializeWithContext(frameContext);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { externalAppCardActions: {} } });
+          const invalidAppId = 'a'.repeat(257);
+          try {
+            externalAppCardActions.processActionOpenUrl(invalidAppId, testUrl);
```

Comment:
Do we not need to `await` here?

Reply:
:( I merged the PR and saw these comments. I will go back and add await.